### PR TITLE
export batoms to dict

### PR DESCRIPTION
**Pull request by superstar54**
In this PR, I added `as_dict()` function to export the  `Batoms` and its submodules to a dictionary.

@alchem0x2A I keep the import function `from_dict()` empty till now, because I think it is related to the new design of `batoms_api`.

### CODE REVIEW

File: batoms/batoms.py [hunk 1]
```diff
@@ -435,6 +435,39 @@ def from_batoms(self, label):
...
+            data (dict): _description_
+
+
+        Returns:
+            Batoms: _description_
+
+        """
+        pass
+
+    def as_dict(self):
```
Comment (alchem0x2A) [inline batoms/batoms.py:453]: looks good. maybe in the next version, we can allow as_dict to output information of any plugins

**Review by alchem0x2A — APPROVED** [review]
I like the way the as_dict and from_dict are constructed. Should be one-step closer to having options controllable in a uniform manner now.

File: batoms/batoms.py (hunk 1 above)
Comment (superstar54) [reply batoms/batoms.py:453]: Yes, I will implement this.

Comment (superstar54) [reply batoms/batoms.py:450]: I made the following changes:
- avoid explicitly loading plugins inside Batoms class,  use all plugins through hooks.
- `as_dict` function export data for activated plugins

Please review.

Comment (alchem0x2A) [reply batoms/batoms.py:450]: lgtm. the plugin hook seems to be the right approach when designing the new version of api.

**Comment by alchem0x2A** [comment]
@superstar54 looks like there is a version conflict of `batoms.py` which you need to cherry pick before merging, but rest seems fine

**Pull request merged**